File: client/src/app/core/requests/UpdateIngredientRequest.ts
import { UnitMeasure } from "../enums/UnitMeasure";

export class UpdateIngredientRequest{

    constructor(

        public id:number,
        public name:string, 
        public price:number, 
        public quantity:number, 
        public measure: UnitMeasure
    ){}
}

File: RecipesApp_/RecipesApp.Core/Requests/UpdateIngredientRequest.cs
using RecipesApp.Common.Enums;
using System;

namespace RecipesApp.Core.Requests
{
    public class UpdateIngredientRequest
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public double Price { get; set; }
        public int Quantity { get; set; }
        public UnitMeasureType Measure { get; set; }
    }
}
